### PR TITLE
Add change_table support to batch multiple changes at once

### PR DESCRIPTION
**Pull request by muffinista**
These changes should combine multiple calls within a `change_table` migration into a single`pt-osc` call. This can really accelerate migrations that involve adding multiple columns to a single large table. For example, as I write this I'm adding ~10 columns to a table with millions of rows. Each `pt-osc` run takes about 30 minutes. Batching them into a single call is a pretty huge speed improvement.

I've been sitting on this code for a week or two because I'd planned on using it to manage a few production deploys before submitting the changes, but I've been unable to do that for unrelated internal org reasons, so I figured I'd go ahead and open a PR.

This is obviously not a small change, and definitely warrants some discussion/thinking, so I understand any reluctance to merge it. Thanks!

### CODE REVIEW

**Comment by sauloperez** [comment]
Wow, that's quite interesting and it does make sense. I'll take a look as soon as I can. Keep in mind though that I can't spend as much time I used to in Departure lately. I'm pretty busy with a ton more things.

**Comment by benlangfeld** [comment]
@muffinista Do you need help getting this prepped? Would be nice to have this in departure.

**Comment by muffinista** [comment]
👍🏼 I'll try and revisit this tomorrow

**Comment by muffinista** [comment]
OK, I've done a bunch of cleanup and refactoring, as well as fix the actual spec so it passes.

File: Gemfile [hunk 1]
```diff
@@ -3,4 +3,5 @@ source 'https://rubygems.org'
 gemspec
 
 gem 'codeclimate-test-reporter', '~> 1.0.3', group: :test, require: nil
+gem 'pry'
```
Comment (guilleiguaran) [inline Gemfile]: Can we remove this change since it seems unrelated to the PR?

**Comment by muffinista** [comment]
Absolutely, it's gone now.

File: lib/active_record/connection_adapters/for_alter.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  def bulk_change_table(table_name, operations) #:nodoc:
+    sqls = operations.flat_map do |command, args|
+      table = args.shift
+      arguments = args
+
+      method = :"#{command}_for_alter"
+
+      raise "Unknown method called : #{method}(#{arguments.inspect})" unless respond_to?(method, true)
+      send(method, table, *arguments)
```
Comment (sauloperez) [inline lib/active_record/connection_adapters/for_alter.rb]: I'd rather keep the encapsulation using `public_send`.

Comment (muffinista) [reply lib/active_record/connection_adapters/for_alter.rb]: Makes sense! I've made that change

**Comment by sauloperez** [comment]
Awesome @muffinista ! thanks for fixing those 9 code climate issues as well!

**Comment by sauloperez** [comment]
I just realized we're missing a new entry in the changelog. Would you mind adding it in a separate PR?

**Comment by muffinista** [comment]
Looks like someone beat me to it! Thanks!
